skrub/_sentence_encoder.py

```diff
@@ -0,0 +1,300 @@
+                "permission to this repo by passing `use_auth_token=<your_token>`"
+            ) from e
+
+        self.input_name_ = sbd.name(column)
+
+        X_out = self._vectorize(column)
+
+        if self.n_components is not None:
+            if (min_shape := min(X_out.shape)) >= self.n_components:
+                self.pca_ = PCA(
+                    n_components=self.n_components,
+                    copy=False,
+                    random_state=self.random_state,
+                )
+                X_out = self.pca_.fit_transform(X_out)
+            else:
+                warnings.warn(
+                    f"The matrix shape is {(X_out.shape)}, and its minimum is "
+                    f"{min_shape}, which is too small to fit a PCA with "
+                    f"n_components={self.n_components}. "
+                    "The embeddings will be truncated by keeping the first "
+                    f"{self.n_components} dimensions instead. "
+                    "Set n_components=None to keep all dimensions and remove "
+                    "this warning."
+                )
+                X_out = X_out[:, : self.n_components]
+
+        self.n_components_ = X_out.shape[1]
+
+        cols = self.get_feature_names_out()
+        X_out = sbd.make_dataframe_like(column, dict(zip(cols, X_out.T)))
+        X_out = sbd.copy_index(column, X_out)
+
+        return X_out
+
+    def transform(self, column):
+        """Transform ``column`` using the SentenceEncoder.
+
+        This method uses the embedding model loaded in memory during ``fit``
+        or ``fit_transform``.
+
+        Parameters
+        ----------
+        column : pandas or polars Series of shape (n_samples,)
+            The string column to compute embeddings from.
+
+        Returns
+        -------
+        X_out : pandas or polars DataFrame of shape (n_samples, n_components)
+            The embedding representation of the input.
+        """
+        check_is_fitted(self, "estimator_")
+
+        if not sbd.is_string(column):
+            raise RejectColumn(f"Column {sbd.name(column)!r} does not contain strings.")
+
+        X_out = self._vectorize(column)
+
+        if hasattr(self, "pca_"):
+            X_out = self.pca_.transform(X_out)
+        elif self.n_components != "all":
+            X_out = X_out[:, : self.n_components]
+
+        cols = self.get_feature_names_out()
+        X_out = sbd.make_dataframe_like(column, dict(zip(cols, X_out.T)))
+        X_out = sbd.copy_index(column, X_out)
+
+        return X_out
+
+    def _vectorize(self, column):
+        is_null = sbd.to_numpy(sbd.is_null(column))
+        column = sbd.to_numpy(column)
+        unique_x, indices_x = unique_strings(column, is_null)
+
+        return self.estimator_.encode(
+            unique_x,
+            normalize_embeddings=False,
+            batch_size=self.batch_size,
+            show_progress_bar=self.verbose,
+        )[indices_x]
```

Comment:
I'm not familiar with SetenceEncoder; who is taking care of moving the embeddings from gpu to numpy on cpu?

Reply:
Sentence-encoder does that for us. It outputs numpy arrays.

Reply:
tha'ts great. maybe we could have a short comment saying that it outputs numpy arrays?